src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1243,6 +1247,35 @@ private void ParseModifiers(SyntaxListBuilder tokens, bool forAccessors)
                             break;
                         }
 
+                    case DeclarationModifiers.File:
+                        // 'file' is only a modifier if it is followed by an optional sequence of modifiers and then by a type keyword.
```

Comment:
>and then by a type keyword [](http://example.com/codeflow?start=108&length=26)

If `file` is followed by a modifier but not a type keyword, should we also consider it a modifier?

Reply:
My opinion is yes. I don't think it's helpful to enforce a modifier ordering here.

Reply:
I think I might have misunderstood this comment 😓. I think we could certainly do this to give a better error if the user tries to use the `file` modifier on a non-type member. It gets a little gnarly in combination with the changes already made in the `features/required-members` branch, so I'd slightly prefer to wait for that feature to merge before trying it. IIRC, we have a bunch of methods which are named things like `IsAsyncOrRequiredAKeyword`. We'd be additionally adding `IsAsyncOrRequiredOrFileAKeyword`..at which point I'd be strongly tempted to start shopping for a better name :)